base/record_replay.h

```diff
@@ -169,6 +170,67 @@ struct CompareMemberByPointerId {
   }
 };
 
+inline unsigned HashInt(uint32_t key) {
+  key += ~(key << 15);
+  key ^= (key >> 10);
+  key += (key << 3);
+  key ^= (key >> 6);
+  key += ~(key << 11);
+  key ^= (key >> 16);
+  return key;
+}
+
+inline unsigned HashInt(uint64_t key) {
+  key += ~(key << 32);
+  key ^= (key >> 22);
+  key += ~(key << 13);
+  key ^= (key >> 8);
+  key += (key << 3);
+  key ^= (key >> 15);
+  key += ~(key << 27);
+  key ^= (key >> 31);
+  return static_cast<unsigned>(key);
+}
+
+// Replay's hashing function for pointers, using the registered pointer id.
+template <typename T>
+struct ReplayPtrHash {
```

Comment:
I think the name here could be better, if it clearly indicates this is for pointers that recordreplay::RegisterPointer was used with.  We already have CompareByPointerId, could this be PointerIdHash ?